gefserver/pier/pier.go

```diff
@@ -176,6 +178,11 @@ func (p *Pier) runJob(job *db.Job, service db.Service, inputPID string) {
 
 	{
 		p.db.SetJobState(job.ID, db.NewJobStateOk("Performing data staging", -1))
+		if len(inputPID) == 0 {
+			p.db.SetJobState(job.ID, db.NewJobStateError("Data staging was canceled, since no input source was provided", 1))
```

Comment:
it's unclear whether only the data staging is cancelled and the job still runs, or if the entire job is cancelled